client/components/seo/twitter-preview/index.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { PropTypes } from 'react';
+import PureRenderMixin from 'react-pure-render/mixin';
+import classNames from 'classnames';
+import crypto from 'crypto';
+
+const baseDomain = url =>
+	url
+		.replace( /^[^/]+[/]*/, '' ) // strip leading protocol
+		.replace( /\/.*$/, '' ); // strip everything after the domain
+
+const blavatarUrl = slug => {
+	const slugHash = crypto
+		.createHash( 'md5' )
+		.update( slug )
+		.digest( 'hex' );
+
+	return `https://secure.gravatar.com/blavatar/${ slugHash }?s=512`;
+};
+
+export const TwitterPreview = React.createClass( {
+	mixins: [ PureRenderMixin ],
+
+	render() {
+		const {
+			url,
+			title,
+			type,
+			description,
+			image,
+			slug
+		} = this.props;
+
+		const previewClasses = classNames( 'twitter-card-preview', {
+			[ `twitter-card-preview__${ type }` ]: true
```

Comment:
when these are aways true, we can add them outside of the object

```js
const previewClasses = classNames(
	'twitter-card-preview',
	`twitter-card-preview__${ type }`
);
```

Of course, as this point, it's all static so you could just put it below as a string

```js
<div className={ `twitter-card-preview twitter-card-preview__${ type }` }>
```